packages/nimble-components/src/wafer-map/modules/event-handler.ts

```diff
@@ -0,0 +1,49 @@
+import type { WaferMap } from '..';
+import type { WaferMapDie, EventHandlerData } from '../types';
+import { ZoomHandler } from './zoom-handler';
+
+/**
+ * EventHandler deals with user interactions and events
+ */
+export class EventHandler {
+    private readonly zoomHandler: ZoomHandler;
+
+    public constructor(private readonly eventHandlerData: EventHandlerData) {
+        this.zoomHandler = new ZoomHandler(eventHandlerData.zoomHandlerData);
+
+        // TODO HoverHandler - initialization
+
+        // TODO HoverHandler -  create a hoverDie element
+    }
+
+    public attachEvents(waferMap: WaferMap): void {
+        waferMap.onmousemove = () => {
+            // TODO HoverHandler - mousemove(e) callback
+        };
+        waferMap.onmouseout = () => {
+            // TODO HoverHandler - mouseout() callback
+        };
+        this.zoomHandler.onZoom = () => {
+            // TODO HoverHandler - toggle hoverDie with false
+
+            this.zoomHandler.reScale();
```

Comment:
Instead of exposing `rescale` publicly, I think it would be cleaner for the zoom handler to expose `onBeforeZoom` and `onAfterZoom` "events". Then the zoom handler could call `rescale` itself in between when those events are fired. That keeps the logic of zooming encapsulated within the zoom handler and limits its API to exposing more predictable lifecycle events.